app/controllers/application_controller.rb

```diff
@@ -5,6 +5,110 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if logged_in? then redirect "/tweets" end
+    erb :"/users/create_user"
+  end
+
+  get '/tweets' do
+    if !logged_in? then redirect "/login" end
+    @user=current_user
+    erb :"/tweets/tweets"
+  end
+
+  get '/tweets/new' do
+    if !logged_in? then redirect "/login" end
+    @user=current_user
+    erb :"/tweets/new"
+  end
+
+  get "/tweets/:id" do
+    if !logged_in? then redirect "/login" end
+   @session=session
+    @tweet = Tweet.find(params[:id])
+    erb :"/tweets/show_tweet"
+  end
+
+  get '/tweets/:id/edit' do
+    if !logged_in? then redirect "/login" end
+    @tweet = Tweet.find(params[:id].to_i)
+    @user=@tweet.user
+    if session[:id]!= @user.id then redirect "/tweets" end
+    erb :"/tweets/edit_tweet"
+  end
+
+  post '/tweets/new' do
+    if !logged_in? then redirect "/login" end
+    if params[:content]=="" then redirect "/tweets/new" end
+    @tweet = Tweet.create(content: params[:content], user_id: session[:id])
+    redirect "/tweets"
+  end
+
+  post '/tweets/:id' do
+    if params[:content]=="" then redirect "/tweets/#{params[:id]}/edit" end
+    @tweet=Tweet.find(params[:id])
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...